x-pack/plugins/transform/public/app/sections/create_transform/components/step_details/step_details_form.tsx

```diff
@@ -93,9 +118,31 @@ export const StepDetailsForm: FC<Props> = React.memo(
     );
     const [transformIds, setTransformIds] = useState<TransformId[]>([]);
     const [indexNames, setIndexNames] = useState<EsIndexName[]>([]);
+
+    // Index pattern state
     const [indexPatternTitles, setIndexPatternTitles] = useState<IndexPatternTitle[]>([]);
     const [createIndexPattern, setCreateIndexPattern] = useState(defaults.createIndexPattern);
+    const [previewDateColumns, setPreviewDateColumns] = useState<string[]>([]);
+    const [indexPatternDateField, setIndexPatternDateField] = useState<string | undefined>(
+      undefined
+    );
+
+    const onTimeFieldChanged = React.useCallback(
+      (e: React.ChangeEvent<HTMLSelectElement>) => {
+        const value = e.target.value;
+        // If the value is an empty string, it's not a valid selection
+        if (value === '') {
+          return;
+        }
+        // Find the time field based on the selected value
+        // this is to account for undefined when user chooses not to use a date field
+        const timeField = previewDateColumns.find((col) => col === value);
 
+        const selectedTimeField = timeField ? timeField : undefined;
```

Comment:
looks like this variable is redundant

Reply:
Fixed in https://github.com/elastic/kibana/commit/3fe458e73e36261939a364c1a1101626b85ba7b1